chainstate/tx-verifier/src/transaction_verifier/input_output_policy/tests.rs

```diff
@@ -0,0 +1,721 @@
+}
+
+fn get_random_outputs_combination(
+    rng: &mut impl Rng,
+    source: &[TxOutput],
+    result_len: usize,
+) -> Vec<TxOutput> {
+    source
+        .iter()
+        .combinations_with_replacement(result_len)
+        .choose(rng)
+        .unwrap()
+        .into_iter()
+        .cloned()
+        .collect::<Vec<_>>()
+}
+
+fn make_block(kernels: Vec<TxInput>, reward_outputs: Vec<TxOutput>) -> Block {
+    let (sk, _) = VRFPrivateKey::new_from_entropy(VRFKeyKind::Schnorrkel);
+    let vrf_data = sk.produce_vrf_data(TranscriptAssembler::new(b"abc").finalize().into());
+    Block::new(
+        vec![],
+        Id::<GenBlock>::new(H256::zero()),
+        BlockTimestamp::from_int_seconds(0),
+        ConsensusData::PoS(Box::new(PoSData::new(
+            kernels,
+            vec![],
+            PoolId::new(H256::zero()),
+            vrf_data,
+            Compact(1),
+        ))),
+        BlockReward::new(reward_outputs),
+    )
+    .unwrap()
+}
+
+fn make_block_no_kernel(reward_outputs: Vec<TxOutput>) -> Block {
+    Block::new(
+        vec![],
+        Id::<GenBlock>::new(H256::zero()),
+        BlockTimestamp::from_int_seconds(0),
+        ConsensusData::None,
+        BlockReward::new(reward_outputs),
+    )
+    .unwrap()
+}
+
+fn prepare_utxos_and_tx(
+    rng: &mut impl Rng,
+    input_utxos: Vec<TxOutput>,
+    outputs: Vec<TxOutput>,
+) -> (UtxosDBInMemoryImpl, Transaction) {
+    let utxos = input_utxos
+        .into_iter()
+        .enumerate()
+        .map(|(i, output)| {
+            (
+                OutPoint::new(
+                    OutPointSourceId::Transaction(Id::new(H256::random_using(rng))),
+                    i as u32,
+                ),
+                Utxo::new_for_mempool(output),
+            )
+        })
+        .collect::<BTreeMap<_, _>>();
+
+    let inputs: Vec<TxInput> = utxos.keys().map(|outpoint| outpoint.clone().into()).collect();
+
+    (
+        UtxosDBInMemoryImpl::new(Id::<GenBlock>::new(H256::zero()), utxos),
+        Transaction::new(0, inputs, outputs, 0).unwrap(),
+    )
+}
+
+fn prepare_utxos_and_tx_with_random_combinations(
+    rng: &mut impl Rng,
+    origin_input_utxos: &[TxOutput],
+    number_of_inputs: usize,
+    origin_outputs: &[TxOutput],
+    number_of_outputs: usize,
+    extra_output: Option<TxOutput>,
+) -> (UtxosDBInMemoryImpl, Transaction) {
+    let input_utxos = get_random_outputs_combination(rng, origin_input_utxos, number_of_inputs);
+
+    let outputs = match extra_output {
+        Some(extra) => get_random_outputs_combination(rng, origin_outputs, number_of_outputs)
+            .into_iter()
+            .chain(std::iter::once(extra))
+            .collect(),
+        None => get_random_outputs_combination(rng, origin_outputs, number_of_outputs),
+    };
+
+    prepare_utxos_and_tx(rng, input_utxos, outputs)
+}
+
+#[rstest]
+    #[rustfmt::skip]
+    #[case(transfer(), transfer(),           Ok(()))]
+    #[case(transfer(), burn(),               Ok(()))]
```

Comment:
Why is the test and all the `#[case]` directives indented? I think the formatter would have fixed it were it not prevented by `#[rustfmt::skip]` but it would also screw up the vertical alignment of the expected results. I suspect a copy-paste error because the module was in the parent file directly before.